### PR TITLE
test: file uploader component e2e-tests 

### PR DESCRIPTION
**Pull request by nikvalor**
#### Please provide a brief summary of this pull request.
Added file uploader component e2e-tests
#### Please check whether the PR fulfills the following requirements

- [ ] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [ ] tests for the changes that have been done
- [ ] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [ ] update `README.md`
- [ ] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [ ] Documentation Examples
- [ ] Stackblitz works for all examples

### CODE REVIEW

File: e2e/wdio/platform/pages/step-input.po.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            driver.keys(['Enter']);
+            checkValueChanged(newValue, webDriver.getValue(this.activeInput, i));
+        }
+    };
+
+    checkInvalidInput(input: string): void {
+        webDriver.scrollIntoView(input);
+        webDriver.clearValue(input);
+        webDriver.setValue(input, 'invalid');
+        driver.keys(['Enter']);
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/pages/step-input.po.ts:84]: please use sendKeys and for other places.

File: e2e/wdio/platform/tests/combobox.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { webDriver } from '../../driver/wdio';
+import { ComboBoxPo } from '../pages/combobox.po';
+import ComboboxData from '../fixtures/appData/combobox.page-content';
+
+describe('Combobox test suite', () => {
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/platform/tests/combobox.e2e-spec.ts:5]: function ?

**Review by AntonOlkhovskyi — CHANGES_REQUESTED** [review]

File: e2e/wdio/driver/wdio.ts [hunk 3]
```diff
@@ -54,19 +54,29 @@ export class Wdio {
     // Clear value before set new
     setValue(selector: string, value: string, index: number = 0, waitTime = this.defaultWaitTime): void {
         $$(selector)[index].waitForDisplayed({ timeout: waitTime });
+        $$(selector)[index].clearValue();
```
[the remark this answers is not in the document]
Comment (nikvalor) [reply e2e/wdio/driver/wdio.ts:57]: It wasn't working in my case

File: e2e/wdio/platform/tests/combobox.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    beforeAll(() => {
+        comboBoxPage.open();
+    });
+
+    afterEach(() => {
+        webDriver.refreshPage();
+    });
+
+    it('Verify each combobox consist of input and button', () => {
+        comboBoxPage.verifyComboBoxComponents(ComboboxData.activeTypeNames, ComboboxData.notActiveTypeNames);
```
Comment (valorkin) [inline e2e/wdio/platform/tests/combobox.e2e-spec.ts:17]: this is over encapsulation, spec files should be readable in terms of `AAA`
where Arange is `open` in 90% of its

Comment (valorkin) [reply e2e/wdio/platform/tests/combobox.e2e-spec.ts:17]: methods should makes sense only if they are used several times mostly

File: e2e/wdio/helper/assertion-helper.ts [hunk 5]
```diff
@@ -1,5 +1,19 @@
 import { webDriver } from '../driver/wdio';
 
+export function checkRtlOrientation(element: string, index: number): void {
+    expect(webDriver.getAttributeByName(element, 'dir', index)).toBe('rtl');
+    expect(webDriver.getCSSPropertyByName(element, 'direction', index).value).toBe('rtl');
+}
+
+export function compareDropDownOptions(firstOptionValue: string, inputOptionValue: string): void {
```
Comment (valorkin) [inline e2e/wdio/helper/assertion-helper.ts:8]: specific to dropdown, in global helper only global methods should be

File: e2e/wdio/platform/pages/step-input.po.ts [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+            webDriver.click(this.activeInput, i);
+            webDriver.sendKeys(['ArrowUp']);
+            webDriver.sendKeys(['Enter']);
+            checkValueChanged(newValue, webDriver.getValue(this.activeInput, i));
+        }
+    };
+
+    checkInvalidInput(input: string): void {
+        webDriver.scrollIntoView(input);
+        webDriver.clearValue(input);
```
Comment (valorkin) [inline e2e/wdio/platform/pages/step-input.po.ts:81]:
```        
webDriver.clearValue(input);
webDriver.setValue(input, 'invalid');
webDriver.sendKeys(['Enter']);
webDriver.waitForDisplayed(this.errorMessage);```

this is a good of reusable code to become method

File: e2e/wdio/platform/pages/step-input.po.ts [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        webDriver.setValue(input, 'invalid');
+        webDriver.sendKeys(['Enter']);
+        webDriver.waitForDisplayed(this.errorMessage);
+        let formStatusText = webDriver.getText(this.formStatusText)
+        checkTextValueContain(formStatusText, 'INVALID')
```
Comment (valorkin) [inline e2e/wdio/platform/pages/step-input.po.ts:86]: `INVALID` should be in spec as `expected` value

**Review by valorkin — CHANGES_REQUESTED** [review]
please make specs more expressive and contain AAA notation
keep `expect`s and expected values in specs not in PO

**Review by Seamoo13 — APPROVED** [review]

**Comment by AntonOlkhovskyi** [comment]
@nikvalor Please reopen using SAP/fundamentals-ngx after comments fix, to make CI work for this PR.

**Comment by nikvalor** [comment]
It was merged as part of https://github.com/SAP/fundamental-ngx/pull/4159/
Closed this one

**Pull request closed without merging**